packages/runtime/container-runtime/src/gc/garbageCollection.ts

```diff
@@ -202,12 +201,8 @@ export class GarbageCollector implements IGarbageCollector {
 					// For newer documents, GC data should be present in the GC tree in the root of the snapshot.
 					const gcSnapshotTree = baseSnapshot.trees[gcTreeKey];
 					if (gcSnapshotTree === undefined) {
-						// back-compat - Older documents will have the GC blobs in each data store's snapshot tree.
-						return getSnapshotDataFromOldSnapshotFormat(
-							baseSnapshot,
-							createParams.metadata,
-							readAndParseBlob,
-						);
+						// back-compat - Older documents need their gc data reset
```

Comment:
Older documents do not _**need**_ their GC data reset but rather we choose to do it for simplicity and since there shouldn't be many old docs that do not have GC data in the root. For them, its fine to regenerate GC data even though its expensive.

Reply:
Also, incremental GC summary won't work with older GC data so that's another reason.